R/reporting_triangle-methods.R

```diff
@@ -0,0 +1,495 @@
+#' @title Subset reporting_triangle objects
+#'
+#' @description
+#' Extract or replace parts of a reporting_triangle object while preserving its
+#' attributes.
+#'
+#' @param x A [reporting_triangle] object
+#' @param ... Indices for subsetting
+#' @return A [reporting_triangle] object with the subset data
+#' @family reporting_triangle
+#' @method [ reporting_triangle
+#' @export
+`[.reporting_triangle` <- function(x, ...) {
```

Comment:
This syntax is new to me, can you explain whats going on here... 
My sense is that this allowed us to do `[ , ]` on the `reporting_triangle`, while retaining it as a `reporting_triangle` object rather than have the new thing being converted back to a matrix? Same with the below for other operations?

Reply:
yes exactly and it will run validaiton to check it is still a reporting triangle without you having to do anything.Same for the allocation one.